agent/consul/state/memdb_wrapper.go

```diff
@@ -42,8 +52,9 @@ func (db *changeTrackerDB) ReadTxn() *txn {
 // data directly into the DB. These cases may use WriteTxnRestore.
 func (db *changeTrackerDB) WriteTxn(idx uint64) *txn {
 	t := &txn{
-		Txn:   db.db.Txn(true),
-		Index: idx,
+		Txn:       db.db.Txn(true),
+		Index:     idx,
```

Comment:
Do we need to pass this index? While I was experimenting I noticed that all of the changes include the latest index in multiple places. The change object has a `ModifyIndex`, and  there is also a change event for the `index` table that appears to contain the latest index.

Could we use those existing indexes in the `TopicHandler.ProcessChanges` functions to set the index on the event, removing the need to pass the index to the Txn for the sole purpose of storing it on the event?